framework/include/utils/RankTwoTensor.h

```diff
@@ -173,85 +281,265 @@ class RankTwoTensorTempl : public TensorValue<T>
   {
   }
 
-  /// Construct from other template
+  /**
+   * @brief The conversion operator from `RankTwoTensorTempl<T2>` to `RankTwoTensorTempl<T>` where
+   * `T2` is convertible to `T`.
+   */
   template <typename T2>
-  RankTwoTensorTempl(const RankTwoTensorTempl<T2> & a) : TensorValue<T>(a)
+  RankTwoTensorTempl(const RankTwoTensorTempl<T2> & a) : libMesh::TensorValue<T>(a)
   {
   }
+  /// @} end Constructor
 
-  // Named constructors
-  [[nodiscard]] static RankTwoTensorTempl Identity() { return RankTwoTensorTempl(initIdentity); }
+  /// @{
 
-  /// Static method for use in validParams for getting the "fill_method"
-  [[nodiscard]] static MooseEnum fillMethodEnum();
+  /**
+   * @brief Named constructor for initializing symmetrically.
+   *
+   * The supplied vectors are used as row and column vectors to construct two tensors respectively,
+   * that are averaged to create a symmetric tensor.
+   *
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.cpp}
+   * RealVectorValue row1(1, 2, 3);
+   * RealVectorValue row2(4, 5, 6);
+   * RealVectorValue row3(7, 8, 9);
+   * auto A = RankTwoTensor::initializeSymmetric(row1, row2, row3);
+   * // A = [ 1 3 5
+   * //       3 5 7
+   * //       5 7 9 ]
```

Comment:
That doesn't look right. all 5s on the antidiagonal? 2 4 6 and 8 don't appear? They should appear as average terms.

Reply:
Oh, I see, you took those as values not as placeholder indices....